torch/_inductor/cudagraph_trees.py

```diff
@@ -416,25 +417,43 @@ def cudagraphify(
     constants: Tuple[torch.Tensor, ...] = (),
     placeholders: Tuple[PlaceholderInfo, ...] = (),
     mutated_input_idxs: Tuple[int, ...] = (),
+    compile_id: Optional[CompileId] = None,
 ) -> Tuple[ModelType, OutputType]:
-    manager = get_container(device_index).get_tree_manager()
     assert not (is_backward and is_inference)
     mode = (
         CompilationMode.BACKWARD
         if is_backward
         else (CompilationMode.INFERENCE if is_inference else CompilationMode.FORWARD)
     )
 
-    return manager.add_function(
-        model,
-        inputs,
-        static_input_idxs,
-        stack_traces,
-        mode,
-        constants,
-        placeholders,
-        mutated_input_idxs,
-    )
+    with dynamo_timed(
```

Comment:
I think you mean so that we can use as a decorator? Unfortunately it looks like there's some unfortunate interaction with dynamo_timed and cProfile when used as a cm. We have this warning: https://github.com/pytorch/pytorch/blob/91bf2e16debdc41f5dde2bb5cc8e4f39f8955d4e/torch/_dynamo/utils.py#L334

Reply:
I could define a new context manager?

```
@contextmanager
def dynamo_timed_cudagraphs(name, compile_id, mode, overhead=False):
    with dynamo_timed(
        name,
        log_pt2_compile_event=True,
        compile_id=compile_id,
        is_forward=mode != CompilationMode.BACKWARD,
       dynamo_compile_runtime_column_us= runtime_cudagraphify_time_us" if overhead else None, 
    ):
        yield
```

That'd make the usages a bit more concise ... wdyt?

Reply:
Sure, sgtm!